### PR TITLE
default value for `zdc` constraints

### PR DESCRIPTION
**Pull request by herbberg**
added default value (module = 0) for `zdc` constraints

### CODE REVIEW

File: changelog [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.14.3] - 2023-09-20
+
+### Added
+- default value (mdule = 0) for `zdc` constraints
+
```
Comment (arnobaer) [inline changelog:14]: In [semantic versioning](https://semver.org/) changing anything but _Fixed_ is not a patch, so go for 2.15.0

File: tmVhdlProducer/main.py [hunk 2]
```diff
@@ -35,6 +35,8 @@
     'ext': [kExternals],
     'zdc': [kZDCPlus, kZDCMinus],
 }
+
+DefaultModuleZdc = 0
 """Mapping constraint types to esCondition types, provided for convenience."""
```
Comment (arnobaer) [inline tmVhdlProducer/main.py:40]: Take care, the doc string was attached to `ConstraintTypes` and is now assigned to `DefaultModuleZdc`!

File: tmVhdlProducer/main.py [hunk 3]
```diff
@@ -184,6 +186,17 @@ def main() -> int:
...
+    # Default constraint for ZDC
+    zdc_constraint = True
+    for key in constraints.keys():
+        if (key == kZDCPlus) or (key == kZDCMinus):
+            zdc_constraint = True
+        else:
+            zdc_constraint = False
+    if (not zdc_constraint) or (len(constraints) == 0):
+        constraints.update( [(kZDCPlus, [DefaultModuleZdc]),(kZDCMinus, [DefaultModuleZdc])] )
+
```
Comment (arnobaer) [inline tmVhdlProducer/main.py:199]: That is a bit complicated, how about this:
```python
# Default constraint for ZDC
constraints.setdefault(kZDCMinus, [DefaultModuleZdc])
constraints.setdefault(kZDCPlus, [DefaultModuleZdc])
```
Also using [list comprehensions](https://realpython.com/list-comprehension-python/#how-to-supercharge-your-comprehensions) would work:
```python
# Default constraint for ZDC
if kZDCMinus not in constraints:
    constraints[kZDCMinus] = [DefaultModuleZdc]
if kZDCPlus not in constraints:
    constraints[kZDCPlus] = [DefaultModuleZdc]
```

File: changelog (hunk 1 above)
Comment (arnobaer) [reply changelog:14]: typo: `mdule` -> `module`

**Review by arnobaer — CHANGES_REQUESTED** [review]

File: changelog (hunk 1 above)
Comment (arnobaer) [reply changelog:14]: And the links table at the bottom is waiting for an update too :) https://github.com/cms-l1-globaltrigger/tm-vhdlproducer/blob/e5f25516a891caa50e5ddeaa4f2e08bc913890cd/changelog#L285-L288

Comment (herbberg) [reply changelog:14]: Done!

File: tmVhdlProducer/main.py (hunk 2 above)
Comment (herbberg) [reply tmVhdlProducer/main.py:40]: Moved "DefaultModuleZdc = 0" up. Done!

**Pull request merged**